src/utils.js

```diff
@@ -12,58 +12,18 @@
  * @singleton
  */
 
-CKEDITOR.define( function() {
-	return {
+CKEDITOR.define( [ 'utils-lodash', 'lib/lodash/lodash-ckeditor' ], function( lodashIncludes, lodash ) {
+	var utils = {
 		/**
-		 * Extends one JavaScript object with the properties defined in one or more objects. Existing properties are
-		 * overridden.
-		 *
-		 * @param {Object} target The object to be extended.
-		 * @param {Object} source One or more objects which properties will be copied (by reference) to `target`.
-		 * @returns {Object} The `target` object.
+		 * The list of methods defined in `utils` that come from lodash.
 		 */
-		extend: function( target, source ) {
-			if ( !this.isObject( source ) ) {
-				return target;
-			}
-
-			if ( arguments.length > 2 ) {
-				var args = Array.prototype.splice.call( arguments, 1 );
-
-				while ( args.length ) {
-					this.extend( target, args.shift() );
-				}
-			} else {
-				var keys = Object.keys( source );
-
-				while ( keys.length ) {
-					var key = keys.shift();
-					target[ key ] = source[ key ];
-				}
-			}
-
-			return target;
-		},
+		_lodashIncludes: lodashIncludes
+	};
 
-		/**
-		 * Checks if the provided object is a JavaScript function.
-		 *
-		 * @param obj The object to be checked.
-		 * @returns {Boolean} `true` if the provided object is a JavaScript function. Otherwise `false`.
-		 */
-		isFunction: function( obj ) {
-			return typeof obj == 'function';
-		},
+	// Extend "utils" with Lo-Dash methods.
+	lodashIncludes.forEach( function( name ) {
```

Comment:
I feel the same way as Fred. God only knows if the performance is really lower (micro benchmarks will not test it) and what's the memory usage, but I never liked `forEach` because of this feeling :D.